ops/framework.py

```diff
@@ -527,10 +546,25 @@ def __init__(self, storage, charm_dir, meta, model):
         # Flag to indicate that we already presented the welcome message in a debugger breakpoint
         self._breakpoint_welcomed = False
 
+        # Define tack observer to tracked history of model config
+        self._tack_observer = ConfigHistoryObserver(self)
+
         # Parse the env var once, which may be used multiple times later
         debug_at = os.environ.get('JUJU_DEBUG_AT')
         self._juju_debug_at = set(x.strip() for x in debug_at.split(',')) if debug_at else set()
 
+    def tacked(self, keys: typing.List[str], depth: int = -1) -> None:
```

Comment:
I wonder if the depth should have a finite default.

Reply:
I've been thinking about it, but I'm not sure what's a reasonable number here.